lua/svartafanan/window.lua

```diff
@@ -34,68 +37,60 @@ function M.open()
 	local row = math.ceil((height - win_height) / 2 - 1)
 	local col = math.ceil((width - win_width) / 2)
 
-	local border_opts = {
-		style = "minimal",
-		relative = "editor",
-		width = win_width + 2,
-		height = win_height + 2,
-		row = row - 1,
-		col = col - 1,
-	}
-
-	local opts = {
+	M.win = api.nvim_open_win(M.buf, true, {
 		style = "minimal",
 		relative = "editor",
 		width = win_width,
 		height = win_height,
 		row = row,
 		col = col,
-	}
+		border = "double",
+	})
 
-	-- Set border buffer lines
-	local border_lines = { "╔" .. string.rep("═", win_width) .. "╗" }
-	local middle_line = "║" .. string.rep(" ", win_width) .. "║"
+	-- fill the buffer with empty lines
+	local empty_lines = {}
 	for _ = 1, win_height do
-		table.insert(border_lines, middle_line)
+		table.insert(empty_lines, "")
 	end
-	table.insert(border_lines, "╚" .. string.rep("═", win_width) .. "╝")
-	api.nvim_buf_set_lines(border_buf, 0, -1, false, border_lines)
-
-	-- Open the border window first then the actual window
-	api.nvim_open_win(border_buf, true, border_opts)
-	local win = api.nvim_open_win(buf, true, opts)
-
-	-- If the window is closed, close the border window as well
-	api.nvim_command('au BufWipeout <buffer> exe "silent bwipeout! "' .. border_buf)
-
-	-- we can add title already here, because first line will never change
-	api.nvim_buf_set_lines(buf, 0, -1, false, { center("SvartaFanan"), "", "" })
-
-	return win, buf -- Return window and buffer handles
+	api.nvim_buf_set_lines(M.buf, 0, -1, false, empty_lines)
+	M.update("SvartaFanan", 1)
+
+	api.nvim_create_autocmd("WinClosed", {
```

Comment:
This listens to the "main" window being closed, right?
So when the window closes - the border "window" also closes att the same time.

Reply:
We don't have a border window anymore. This autocmd is triggered if the svartafanan window gets closed and cleans up after the plugin (closes buffer, stops timer).

We need this autocmd because if the window is closed, the timer will continue to try to write to buf, which is no longer visible (or deleted).